### PR TITLE
chore: replace usage of `Directive::Quotient` with brillig opcode 

### PR DESCRIPTION
**Pull request by guipublic**
# Description

Use brillig opcode instead of the quotient directive

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

I will mark the PR as ready once the PRs for inverse directive and truncation removal are merged.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
For reference, putting the quotient directive here: https://github.com/noir-lang/acvm/blob/7d28f42c71d8434e73dfa69b94257c00ef7b6036/acvm/src/pwg/directives/mod.rs#L48

This uses BigUint, which is not a native type in Brillig. What steps,if any, are we taking to make this work?

**Comment by kevaundray** [comment]
Related to #1772

**Comment by guipublic** [comment]
> For reference, putting the quotient directive here: https://github.com/noir-lang/acvm/blob/7d28f42c71d8434e73dfa69b94257c00ef7b6036/acvm/src/pwg/directives/mod.rs#L48
> 
> This uses BigUint, which is not a native type in Brillig. What steps,if any, are we taking to make this work?

We do not use BigUint here, we have the bit size of the inputs and so we simply use brillig integer division.

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs [hunk 1]
```diff
@@ -357,6 +348,29 @@ impl GeneratedAcir {
         Ok((q_witness, r_witness))
     }
 
+    pub(crate) fn brillig_quotient(
+        &mut self,
+        lhs: &Expression,
+        rhs: &Expression,
+        predicate: &Expression,
+        max_bit_size: u32,
```
Comment (sirasistant) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:356]: In the brillig_vm, the maximum bit_size for binary operations that we support is currently 127 bits, is it guaranteed that the quotient max_bit_size will never exceed 127 bits?

Comment (guipublic) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:356]: No, it is not guaranteed, you should support any bit size.

**Comment by kevaundray** [comment]
@guipublic what is the status of this PR?

**Review by TomAFrench — DISMISSED** [review]
This should be good to go as the relevant ACVM PR is now being used.

**Comment by TomAFrench** [comment]
@guipublic are we good to merge this?

**Comment by guipublic** [comment]
> @guipublic are we good to merge this?

Yes

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**